atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTopologyValidatorTest.java

```diff
@@ -278,6 +278,143 @@ public void validateNewlyAddedHostsNewHostsNotAddedIfTheyDisagreeWithOldHostsOnP
                 .containsExactlyElementsOf(newCassandraServers);
     }
 
+    @Test
+    public void returnsAllNewHostsIfNoConsensusOnNewHostsWhenNoQuorumAndCurrentServersExist() {
+        Iterator<String> uuidIterator = UUIDS.iterator();
+        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = setupHosts(ALL_HOSTS);
+        Map<CassandraServer, CassandraClientPoolingContainer> oldHosts =
+                filterServerToContainerMap(allHosts, OLD_HOSTS::contains);
+        Set<CassandraServer> oldCassandraServers = oldHosts.keySet();
+        Set<CassandraServer> newCassandraServers = filterServers(allHosts, NEW_HOSTS::contains);
+
+        setHostIds(filterContainers(allHosts, OLD_HOSTS::contains), HostIdResult.success(UUIDS));
+        validator.getNewHostsWithInconsistentTopologies(oldCassandraServers, oldHosts);
+
+        Set<String> hostsOffline = OLD_HOSTS;
+        setHostIds(filterContainers(allHosts, hostsOffline::contains), HostIdResult.hardFailure());
+        filterContainers(allHosts, server -> !hostsOffline.contains(server))
+                .forEach(container ->
+                        setHostIds(ImmutableSet.of(container), HostIdResult.success(Set.of(uuidIterator.next()))));
+
+        assertThat(hostsOffline).hasSizeGreaterThanOrEqualTo((ALL_HOSTS.size() + 1) / 2);
```

Comment:
The goal was to assert that we're not just rejecting all the hosts because there's no overall quorum anyway (that is, this assert is to ensure that we don't have a quorum of successes)